common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt

```diff
@@ -0,0 +1,227 @@
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+
+        ViewTreeLifecycleOwner.get(this)?.lifecycle?.removeObserver(viewModel)
+
+        viewModel.removeNotifyMeListener()
+
+        listener = null
+
+        coroutineScope?.cancel()
+        coroutineScope = null
+    }
+
+    fun setListener(listener: NotifyMeListener?) {
+        this.listener = listener
+    }
+
+    fun setTitle(title: String) {
+        binding.notifyMeMessageTitle.text = title
+    }
+
+    fun setSubtitle(subtitle: String) {
+        binding.notifyMeMessageSubtitle.text = subtitle
+    }
+
+    private fun render(viewState: ViewState) {
+        if (viewState.visible) {
+            showMe()
+        } else {
+            hideMe()
+        }
+    }
+
+    private fun processCommands(command: Command) {
+        when (command) {
+            is CheckPermissions -> checkNotificationsPermissions()
+            is OpenSettings -> openSettings()
+            is Close -> hideMe()
+            is CheckShouldShowRequestPermissionRationale -> checkRequestPermissionRationale()
+            is ShowPermissionRationale -> showNotificationsPermissionsPrompt()
+        }
+    }
+
+    private fun checkNotificationsPermissions() {
+        if (isAtLeastAndroid13()) {
+            val value =
+                ContextCompat.checkSelfPermission(context, Manifest.permission.POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED
+            viewModel.updateNotificationsPermissions(value)
+        } else {
+            viewModel.updateNotificationsPermissions(true)
+        }
+    }
+
+    private fun openSettings() {
+        // This gets called only on Android 13+.
+        @SuppressLint("InlinedApi")
+        val settingsIntent = Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS)
+            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            .putExtra(Settings.EXTRA_APP_PACKAGE, context.packageName)
+
+        startActivity(context, settingsIntent, null)
+    }
+
+    private fun showMe() {
+        this.show()
+    }
+
+    private fun hideMe() {
+        this.gone()
+    }
+
+    private fun checkRequestPermissionRationale() {
+        // This gets called only on Android 13+.
+        @SuppressLint("InlinedApi")
+        val showPrompt = ActivityCompat.shouldShowRequestPermissionRationale(getActivity(), Manifest.permission.POST_NOTIFICATIONS)
+        if (showPrompt) {
+            viewModel.onPermissionRationaleNeeded()
+        } else {
+            viewModel.onOpenSettings()
+        }
+    }
+
+    private fun showNotificationsPermissionsPrompt() {
+        if (isAtLeastAndroid13()) {
+            ActivityCompat.requestPermissions(getActivity(), arrayOf(Manifest.permission.POST_NOTIFICATIONS), 1)
+        }
+    }
+
+    @ChecksSdkIntAtLeast(api = Build.VERSION_CODES.TIRAMISU)
+    private fun isAtLeastAndroid13(): Boolean {
+        return Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU
```

Comment:
use `AppBuildConfig` instead. If you move to `Events`, then the dependency can be added to the `ViewModel` and the `View` has less logic.